Network/DNS/Decode/Internal.hs

```diff
@@ -240,9 +239,12 @@ getDomain' sep1 = do
             let offset = n * 256 + d
             mo <- pop offset
             case mo of
-                Nothing -> fail $ "getDomain: " ++ show offset
-                -- A pointer may refer to another pointer.
-                -- So, register this position for the domain.
+                Nothing -> do
+                    inp <- getInput
+                    case runSGet getDomain (B.drop offset inp) of
```

Comment:
In the very unlikely case that an SOA `mrname` is a pointer to another domain, should the `getDomain` on this line be `(getDomain' sep1)`?

Reply:
Also what prevents infinite pointer chasing recursion in maliciously constructed reply packets?

Reply:
> In the very unlikely case that an SOA `mrname` is a pointer to another domain, should the `getDomain` on this line be `(getDomain' sep1)`?

Right. Fixed.

Reply:
> Also what prevents infinite pointer chasing recursion in maliciously constructed reply packets?

c27ef782013e53cdfda055b3af29aa4f7fc40190 should fix this.